scripts/react_native_pods.rb

```diff
@@ -192,7 +192,7 @@ def use_react_native_codegen!(spec, options={})
     :name => 'Generate Specs',
     :input_files => [srcs_dir],
     :output_files => ["$(DERIVED_FILE_DIR)/codegen-#{codegen_modules_library_name}.log"].concat(generated_files),
-    :script => "set -o pipefail\n\nbash -l -c '#{env_vars} CODEGEN_MODULES_LIBRARY_NAME=#{codegen_modules_library_name} #{File.join(__dir__, "generate-specs.sh")}' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
+    :script => "set -o pipefail\n\nbash -l -c '#{env_vars} CODEGEN_MODULES_LIBRARY_NAME=#{codegen_modules_library_name} ../../node_modules/react-native/scripts/generate-specs.sh' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
```

Comment:
This is just a question to showcase my ignorance: I have fixed this locally by using `SRCROOT` which will resolve to an absolute path at runtime, because I was under the impression that would be the more 'canonical' way. Do you think it is correct to do so (my build passes, so that's at least 1 criterion) or is it better to have it relative like you have done?

Reply:
Is this `SRCROOT` something that is already defined by the build process (and does it point to the same absolute directory in the prepare command and script phase)? If so, that would definitely be better in my opinion; I don't like this relative addressing at all (I just don't know enough about this build system to suggest a good fix for the issue).

Reply:
It's pretty obscure to me too, luckily in the logs on our build server all environment variables are printed at one point. Here is a small excerpt of ones that I think could be relevant in this context:
```
export PODS_ROOT\=/Users/runner/work/1/s/ios/Pods
export PODS_TARGET_SRCROOT\=/Users/runner/work/1/s/ios/Pods/../../node_modules/react-native/React/FBReactNativeSpec
export SOURCE_ROOT\=/Users/runner/work/1/s/ios/Pods
export SRCROOT\=/Users/runner/work/1/s/ios/Pods
```
There is some overlap here and my initial googling hasn't exactly given me any pointers as to which ones are future proof. I actually ended up using `PODS_ROOT` for the prefix, because on my machine `SRCROOT` turned out to be pointing to `/`, which gave me permission errors.

Anyway, here is some documentation as well, which could potentially proof helpful: https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html.

Reply:
I tried to test this myself and verified that I have similar variables properly set in the script phase. However, when prepare command is run (`pod install`) all the variables I tested were unset, so the paths would be incorrect in that phase if the paths are based on those variables.